Azure.ResourceManager.Core/SubscriptionOperations.cs

```diff
@@ -57,8 +58,27 @@ protected SubscriptionOperations(SubscriptionOperations subscription, ResourceId
         /// </summary>
         /// <param name="resourceGroupName"> The name of the resource group. </param>
         /// <returns> The resource group operations. </returns>
+        /// <exception cref="ArgumentNullException"> ResourceGroupName cannot be null or empty. </exception>
+        /// <exception cref="ArgumentException"> ResourceGroupName cannot be longer than 90 characters. </exception>
+        /// <exception cref="ArgumentException"> ResourceGroupName cannot have a special character. </exception>
         public ResourceGroupOperations GetResourceGroupOperations(string resourceGroupName)
         {
+            if (string.IsNullOrWhiteSpace(resourceGroupName))
```

Comment:
I actually prefer if the validation of what a good resourceGroupName is belonged to the resourceGroupOperations class instead.  Your tests will still work as they are written but the exceptions would be thrown from the constructor of ResourceGroupOperations